Field-Robotics-Lab/dave
# Add new models from Ignition Fuel

Builds on top of #194
Demonstrates an example for #191

Adds new models (`Lionfish`, `Niskin`, `Sunken Vase`, `Sunken Ship`) Cole uploaded to the [DAVE collection](https://app.ignitionrobotics.org/Cole/fuel/collections/DAVE) on Ignition Fuel.

They took a few iterations of troubleshooting. `Sunken Vase 02` is commented out for now because it throws an OGRE error, and we're not sure why.

Inertial values are from MeshLab. I'm not planning to add inertia for the ship and the fish, since they're mostly for decoration and maybe for sensor scanning, not any kind of physical interaction. (At least... I wasn't told we were planning to do any deep sea fishing.)

To test:
```
roslaunch dave_demo_launch dave_models.launch
```

You should see messages printed about downloading the models from Ignition Fuel:
```
[Msg] Downloaded model URL: 
          https://fuel.ignitionrobotics.org/1.0/Cole/models/Sunken Vase
      to: 
...
```

They should show up like this (3 objects on the far right and the big ship):
Here I started Gazebo paused to show their initial poses. When unpaused, they fall to the ground and won't look upright.
![2022-02-19-022811_1955x1183_scrot_SunkenShip](https://user-images.githubusercontent.com/7608908/154791521-58433513-e08c-4eb6-b8d8-ef22d176d4f1.png)
![2022-02-19-022819_1955x1183_scrot](https://user-images.githubusercontent.com/7608908/154791687-d95026ac-66cd-43c9-93dd-43cabaf8526f.png)

Inertia for Sunken Vase and Niskin:
![2022-02-19-053157_3706x2049_scrot](https://user-images.githubusercontent.com/7608908/154797236-dc5275f9-48e5-4d36-83ec-6f32e6bdc768.png)
